src/H5public.h

```diff
@@ -217,7 +217,14 @@ typedef int herr_t;
 typedef bool hbool_t;
 typedef int  htri_t;
 
-/* Define the ssize_t type if it not is defined */
+/* The signed version of size_t
+ *
+ * ssize_t is POSIX and not defined in any C standard. It's used in some
+ * public HDF5 API calls so this work-around will define it if it's not
+ * present.
+ *
+ * Use of ssize_t should be discouraged in new code.
+ */
```

Comment:
Intentionally not a Doxygen comment since this is a work-around vs a new HDF5 type.

Reply:
Why not do something similar to `h[s]size_t`, i.e., have a description for users and then an `\internal` elaboration?

Reply:
Because ssize_t is a POSIX type and people can google for its purpose. We don't need to Doxygen up wrappers that paper over platform inconsistencies.